### PR TITLE
Envest/collect mutations

### PR DESCRIPTION
**Pull request by envest**
This script produces four mutation-related output files based on the public TCGA MC3 MAF (which as far as I can tell reports coding SNVs and indels). We focus on BRCA and GBM primary solid tumors, and mutations from PIK3CA, PTEN, and TP53.

- `data/mutations.BRCA.tsv` and `data/mutations.GBM.tsv`
  - BRCA- or GBM-specific data frame with 0/1 for if a sample has a mutation in a gene
  - column names: tcga_id, PIK3CA_mutation, PTEN_mutation, TP53_mutation
  - This is what we can could use as input to machine learning models
- `data/mutations.maf`
  - Full MAF record of all mutations matching our criteria: from BRCA/GBM, primary solid tumors, and our genes of interest 
- `data/mutations.tsv`
  - Pared-down info about all mutations matching our criteria: see directly above

A few lines from `data/mutations.BRCA.tsv`. These 0/1 column vectors will serve as input to ML steps.
| tcga_id | PIK3CA_mutation | PTEN_mutation | TP53_mutation |
| - | - | - | - |
| TCGA-3C-AAAU-01 | 0	| 0 | 0 |
| TCGA-3C-AALI-01 | 0 | 0 | 1 | 
| TCGA-3C-AALJ-01 | 0 | 0 | 0 |
| TCGA-3C-AALK-01 | 1 | 0 | 0 |

A few lines from `data/mutations.tsv`. I think this and the MAF output will be useful for mutation QC to ensure we are using good mutations as the basis of our mutation status inputs to ML steps.
| tcga_id | cancer_type | Tumor_Sample_Barcode | Matched_Norm_Sample_Barcode | Hugo_Symbol | Chromosome | Start_Position | HGVSc | VARIANT_CLASS |
| - | - | - | - | - | - | - | - | - |
TCGA-02-0003-01 | GBM | TCGA-02-0003-01A-01D-1490-08 | TCGA-02-0003-10A-01D-1490-08 | TP53 | 17 | 7577094 | c.844C>T | SNV |
TCGA-02-0003-01 | GBM | TCGA-02-0003-01A-01D-1490-08 | TCGA-02-0003-10A-01D-1490-08 | TP53 | 17 | 7578396 | c.534C>A | SNV |
TCGA-02-0033-01 | GBM | TCGA-02-0033-01A-01D-1490-08 | TCGA-02-0033-10A-01D-1490-08 | TP53 | 17 | 7577538 | c.743G>A | SNV |
TCGA-02-0033-01 | GBM | TCGA-02-0033-01A-01D-1490-08 | TCGA-02-0033-10A-01D-1490-08 | PIK3CA | 3 | 178936082 | c.1624G>A | SNV |

I'm still reviewing these mutations for QC and may still propose a filter or two when trawling the MC3. I felt that python's `optparse` would be overkill in this situation but we can implement that if necessary.

### CODE REVIEW

File: retrieve_MC3_mutations.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    # add TCGA ID to mutation dict
+    if tcga_id not in mutation_dict[cancer_type]:
+      mutation_dict[cancer_type][tcga_id] = []
+
+    # if gene of interest, add to mutation dict list for that ID and outputs
+    if hugo_symbol in genes_of_interest:
+      chromosome = record[4]
+      start_position = record[5]
+      hgvsc = record[34]
+      variant_class = record[94]
```
Comment (jaclyn-taroni) [inline retrieve_MC3_mutations.py:76]: Comments here please!

**Review by jaclyn-taroni — APPROVED** [review]
This seems fine to me and I assume is written the way that it is because the MAF files are very large (because they contain data for every cancer type). If you want a _particularly thorough_ review with an eye towards possible improvements to `retrieve_MC3_mutations.py`, we should probably ask someone better versed in python than me.

**Comment by cgreene** [comment]
I think @jjc2718 has dealt with a fair bit of this logic if you did want 👀

File: retrieve_MC3_mutations.py (hunk 1 above)
Comment (envest) [reply retrieve_MC3_mutations.py:76]: Thanks @jaclyn-taroni -- will add some comments!

**Comment by envest** [comment]
Happy for @jjc2718 to take a look if interested :) no rush, and no expectation implied!

**Review by jjc2718 — COMMENTED** [review]
This looks good! I took a look at `retrieve_MC3_mutations.py` and made some suggestions. None of them are critical (in other words, the script seems like it should work fine as is, my comments are mostly related to readability and reusability), so feel free to create a separate issue/PR to make improvements down the road if you prefer.

As a general comment, Greg wrote some code for doing a similar thing [here in the `greenelab/pancancer` repo](https://github.com/greenelab/pancancer/blob/master/scripts/initialize/process_sample_freeze.py) (and I've been using his data), so you could look at that if you're interested in what we've done for this before. He filtered to certain types of non-silent mutations - not sure if you want to do that too, or if you're okay with using any mutation, which I think is what you're doing currently.

**Comment by envest** [comment]
Thanks, @jjc2718! I think these are really great comments and will go a long way to improving readability/reusability. I'm creating an issue to refer back to these comments and would really like to implement your suggestions in the near future.

**Pull request merged**